### PR TITLE
msgr: set close on exec flag

### PR DESCRIPTION
**Pull request by tchaikov**
mds execv() when handling the "respawn" command, to avoid fd leakage,
and enormous CLOSE_WAIT connections after respawning, we need to set
FD_CLOEXEC flag for the socket fds.

Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

**Comment by yuyuyu101** [comment]
lgtm

File: src/msg/simple/Accepter.cc [hunk 1]
```diff
@@ -63,6 +63,11 @@ int Accepter::bind(const entity_addr_t &bind_addr, const set<int>& avoid_ports)
     return -errno;
   }
 
+  if (set_close_on_exec(listen_sd)) {
```
Comment (tchaikov) [inline src/msg/simple/Accepter.cc]: please move this change to another commit.

Comment (tchaikov) [inline src/msg/simple/Accepter.cc]: sure.

**Comment by tchaikov** [comment]
@ukernel do you think if mds "respawn" is an unusual case or not? if yes, i will file a tracker ticket and backport this changeset to jewel after it gets merged.

**Comment by tchaikov** [comment]
rebased against master

**Comment by tchaikov** [comment]
tested at 

* http://pulpito.ceph.com/kchai-2016-07-02_20:22:26-rados-wip-kefu-testing---basic-smithi/
* http://pulpito.ceph.com/kchai-2016-07-02_09:03:22-rados-wip-kefu-testing---basic-mira/
